src/seraphis_main/contextual_enote_record_types.cpp

```diff
@@ -140,6 +249,43 @@ const SpEnoteSpentContextV1& spent_context_ref(const ContextualRecordVariant &va
     return variant.visit(visitor());
 }
 //-------------------------------------------------------------------------------------------------------------------
+bool is_older_than(const LegacyEnoteOriginContextV1 &context, const LegacyEnoteOriginContextV1 &other_context)
+{
+    // 1. origin status (higher statuses are assumed to be 'older')
+    if (context.origin_status > other_context.origin_status)
+        return true;
+    if (context.origin_status < other_context.origin_status)
+        return false;
+
+    // 2. block index
+    if (context.block_index < other_context.block_index)
+        return true;
+    if (context.block_index > other_context.block_index)
+        return false;
+
+    // note: don't assess the tx output index
+
+    // 3. enote same amount ledger index
+    if (context.enote_same_amount_ledger_index < other_context.enote_same_amount_ledger_index)
```

Comment:
I have a couple thoughts on this here.. also pinging @UkoeHB on this 

First, it appears `is_older_than` is used to update the state of "found enotes" and their origin contexts ([source](https://github.com/UkoeHB/monero/blob/7c80674a56b3eb1efe9847d56ddaf04b006cd0cc/src/seraphis_main/contextual_enote_record_utils.cpp#L233-L244)). @UkoeHB, why is "older" relevant vs actual chain state? Why only update an enote's origin context if and only if the scanner is picking up an **older** origin context? For example, what if a reorg happens and an enote ends up in a higher indexed block .. technically we'd want the *younger* origin context saved.. I don't see why a rule that older always wins makes sense, aside from the obvious case of offchain -> pool -> on chain

Assuming this `is_older_than` should remain, it appears it's currently used to compare potentially 2 different origin contexts **of the same enote**, and not against a different enote. This is important to note because otherwise this function would be incorrect since an enote of transparent amount X can have a higher index for its amount versus enote of transparent amount Y, even though enote Y might appear earlier in the chain than enote X. Maybe an assert that tx id is the same and a comment explaining this function should only be used to compare the same 2 enotes could help for this function.

Reply:
> Using the older one is defensive against reorgs that remove copies of a duplicate.

Can you expand on this scenario?

> Isn't this only true for the amount-based index, not the global enote index?

Yep, that last paragraph is relevant strictly for the `is_older_than` as written in this PR

Reply:
Got it. It seems this `is_older_than` in this PR would need to know the tx's index in a block (in addition to the `enote_index` in the tx) to say definitively which pre-RCT enote is older, if 2 pre-RCT enotes are in the same block.

Reply:
Regardless, since it's possible to have duplicate onetime addresses with different transparent amounts, comparing `enote_same_amount_ledger_index` in this function in this PR wouldn't work